crates/vm/levm/src/vm.rs

```diff
@@ -907,53 +844,191 @@ impl VM {
+        &mut self,
+        address: Address,
+        new_bytecode: Bytes,
+    ) -> Result<(), VMError> {
+        let account = self.get_account_mut(address)?;
+        account.info.bytecode = new_bytecode;
+        Ok(())
+    }
+
+    pub fn update_account_storage(
+        &mut self,
+        address: Address,
+        key: H256,
+        new_value: U256,
+    ) -> Result<(), VMError> {
+        let account = self.get_account_mut(address)?;
+        let account_original_storage_slot_value = account
+            .storage
+            .get(&key)
+            .map_or(U256::zero(), |slot| slot.original_value);
+        let slot = account.storage.entry(key).or_insert(StorageSlot {
+            original_value: account_original_storage_slot_value,
+            current_value: new_value,
+        });
+        slot.current_value = new_value;
+        Ok(())
+    }
+
+    pub fn get_account_mut(&mut self, address: Address) -> Result<&mut Account, VMError> {
+        if !cache::is_account_cached(&self.cache, &address) {
+            let account_info = self.db.get_account_info(address);
+            let account = Account {
+                info: account_info,
+                storage: HashMap::new(),
+            };
+            cache::insert_account(&mut self.cache, address, account.clone());
         }
+        cache::get_account_mut(&mut self.cache, &address)
+            .ok_or(VMError::Internal(InternalError::AccountNotFound))
     }
 
-    /// Gets storage slot, first checking the cache and then the database (caching in the second case)
-    pub fn get_storage_slot(&mut self, address: &Address, key: H256) -> StorageSlot {
-        match self.cache.get_storage_slot(*address, key) {
-            Some(slot) => slot,
+    /// Gets account, first checking the cache and then the database (caching in the second case)
+    pub fn get_account(&mut self, address: Address) -> Account {
+        match cache::get_account(&self.cache, &address) {
+            Some(acc) => acc.clone(),
             None => {
-                let value = self.db.get_storage_slot(*address, key);
-                let slot = StorageSlot {
-                    original_value: value,
-                    current_value: value,
+                let account_info = self.db.get_account_info(address);
+                let account = Account {
+                    info: account_info,
+                    storage: HashMap::new(),
                 };
-                let mut acc = self.get_account(address);
-                acc.storage.insert(key, slot.clone());
-                self.cache.add_account(address, &acc);
-                slot
+                cache::insert_account(&mut self.cache, address, account.clone());
+                account
             }
         }
     }
+
+    // /// Gets storage slot, first checking the cache and then the database (caching in the second case)
+    // pub fn get_storage_slot(&mut self, address: Address, key: H256) -> StorageSlot {
```

Comment:
Should we delete this?

Reply:
Yes! My bad.

Reply:
Done in [c83a496](https://github.com/lambdaclass/ethrex/pull/1314/commits/c83a4964132f911d081fc4a5ab8bbf71d49480ff)